Panic when deleting expired items

Related codes

```go
cacheStore := cache.New(cache.AsLFU[string, SomeStruct](lfu.WithCapacity(5)))
cacheStore.Set(someKey, someValue, cache.WithExpiration(10 * time.Second))
```

Logs

```
panic: runtime error: index out of range [-1]

goroutine 71 [running]:
github.com/Code-Hex/go-generics-cache/policy/lfu.priorityQueue[...].Swap(...)
        github.com/Code-Hex/go-generics-cache@v1.2.1/policy/lfu/priotiry_queue.go:51
container/heap.Remove({0x1e0c8c0, 0xc0006982b8}, 0xffffffffffffffff)
        container/heap/heap.go:72 +0x53
github.com/Code-Hex/go-generics-cache/policy/lfu.(*Cache[...]).Delete(0xc0006982d0, {0xc000322181, 0xbf})
        github.com/Code-Hex/go-generics-cache@v1.2.1/policy/lfu/lfu.go:92 +0x65
github.com/Code-Hex/go-generics-cache.(*Cache[...]).DeleteExpired(0xc0002c2a20)
        github.com/Code-Hex/go-generics-cache@v1.2.1/cache.go:209 +0x158
github.com/Code-Hex/go-generics-cache.NewContext[...].func1()
        github.com/Code-Hex/go-generics-cache@v1.2.1/cache.go:175 +0x25
github.com/Code-Hex/go-generics-cache.(*janitor).run.func1()
        github.com/Code-Hex/go-generics-cache@v1.2.1/janitor.go:39 +0x122
created by github.com/Code-Hex/go-generics-cache.(*janitor).run
        github.com/Code-Hex/go-generics-cache@v1.2.1/janitor.go:33 +0x72
```

--- a/policy/lfu/lfu.go
+++ b/policy/lfu/lfu.go
@@ -2,6 +2,7 @@ package lfu
 
 import (
 	"container/heap"
+	"log"
 )
 
 // Cache is used a LFU (Least-frequently used) cache replacement policy.
@@ -69,6 +70,8 @@ func (c *Cache[K, V]) Set(key K, val V) {
 
 	if len(c.items) == c.cap {
 		evictedEntry := heap.Pop(c.queue).(*entry[K, V])
+		// e := evictedEntry
+		// log.Println(c.items, e.index, e.key)
 		delete(c.items, evictedEntry.key)
 	}
 
@@ -89,6 +92,7 @@ func (c *Cache[K, V]) Keys() []K {
 // Delete deletes the item with provided key from the cache.
 func (c *Cache[K, V]) Delete(key K) {
 	if e, ok := c.items[key]; ok {
+		log.Println(c.items, e.index, e.key)
 		heap.Remove(c.queue, e.index)
 		delete(c.items, key)
 	}

--- a/policy/lfu/lfu_test.go
+++ b/policy/lfu/lfu_test.go
@@ -44,7 +44,7 @@ func TestSet(t *testing.T) {
 }
 
 func TestDelete(t *testing.T) {
-	cache := lfu.NewCache[string, int](lfu.WithCapacity(1))
+	cache := lfu.NewCache[string, int](lfu.WithCapacity(2))
 	cache.Set("foo", 1)
 	if got := cache.Len(); got != 1 {
 		t.Fatalf("invalid length: %d", got)
@@ -62,4 +62,16 @@ func TestDelete(t *testing.T) {
 	if _, ok := cache.Get("foo"); ok {
 		t.Fatalf("invalid get after deleted %v", ok)
 	}
+}
+
+// check don't panic
+func TestIssue33(t *testing.T) {
+	cache := lfu.NewCache[string, int](lfu.WithCapacity(2))
+	cache.Set("foo", 1)
+	cache.Set("foo2", 2)
+	cache.Set("foo3", 3)
+
+	cache.Delete("foo")
+	cache.Delete("foo2")
+	cache.Delete("foo3")
 }

--- a/policy/lfu/priotiry_queue.go
+++ b/policy/lfu/priotiry_queue.go
@@ -65,7 +65,11 @@ func (l *priorityQueue[K, V]) Pop() interface{} {
 	entry := old[n-1]
 	old[n-1] = nil   // avoid memory leak
 	entry.index = -1 // for safety
-	*l = old[0 : n-1]
+	new := old[0 : n-1]
+	for i := 0; i < len(new); i++ {
+		new[i].index = i
+	}
+	*l = new
 	return entry
 }
 

